nix/plugin-overlay.nix

```diff
@@ -1,9 +1,68 @@
 {
   name,
   self,
-}: final: prev: {
-  vimPlugins.rocks-nvim = final.pkgs.vimUtils.buildVimPlugin {
-    inherit name;
+}: final: prev: let
+  lib = final.lib;
+  rocks-nvim-luaPackage-override = luaself: luaprev: {
+    rocks-nvim = luaself.callPackage ({
+      luaOlder,
+      buildLuarocksPackage,
+      lua,
+      toml,
+      toml-edit,
+    }:
+      buildLuarocksPackage {
+        pname = name;
+        version = "scm-1";
+        knownRockspec = "${self}/rocks.nvim-scm-1.rockspec";
+        src = self;
+        disabled = luaOlder "5.1";
+        propagatedBuildInputs = [toml toml-edit];
+      }) {};
+  };
+  lua5_1 = prev.lua5_1.override {
+    packageOverrides = rocks-nvim-luaPackage-override;
+  };
+  lua51Packages = final.lua5_1.pkgs;
+in {
+  inherit lua5_1 lua51Packages;
+
+  vimPlugins.rocks-nvim = final.neovimUtils.buildNeovimPlugin {
+    pname = name;
+    version = "dev";
     src = self;
   };
+
+  neovim-with-rocks = let
+    neovimConfig = final.neovimUtils.makeNeovimConfig {
+      withPython3 = true;
+      viAlias = true;
+      vimAlias = true;
+      plugins = with final.vimPlugins; [
+        rocks-nvim
+      ];
+    };
+    runtimeDeps = with final; [
+      lua5_1
+      luarocks
+    ];
+  in
+    final.wrapNeovimUnstable final.neovim-nightly (neovimConfig
```

Comment:
pretty cool. Hopefully ithe interface will become easier after 23.11. I prefer to keep the wrapperArgs as a list and was thinking of making it mandatory in next nixpkgs (but I can't justify it yet so maybe wont).